File: qc-core/src/main/java/com/eager/questioncloud/core/domain/cart/infrastructure/repository/CartItemRepository.java
package com.eager.questioncloud.core.domain.cart.infrastructure.repository;

import com.eager.questioncloud.core.domain.cart.dto.CartItemDetail;
import com.eager.questioncloud.core.domain.cart.model.CartItem;
import java.util.List;

public interface CartItemRepository {
    CartItem save(CartItem cartItem);

    List<CartItemDetail> findByUserId(Long userId);

    void deleteByIdInAndUserId(List<Long> ids, Long userId);

    Boolean isExistsInCart(Long userId, Long questionId);
}

File: qc-core/src/main/java/com/eager/questioncloud/core/domain/cart/infrastructure/repository/CartItemRepository.kt
package com.eager.questioncloud.core.domain.cart.infrastructure.repository

import com.eager.questioncloud.core.domain.cart.dto.CartItemDetail
import com.eager.questioncloud.core.domain.cart.model.CartItem

interface CartItemRepository {
    fun save(cartItem: CartItem): CartItem

    fun findByUserId(userId: Long): List<CartItemDetail>

    fun deleteByIdInAndUserId(ids: List<Long>, userId: Long)

    fun isExistsInCart(userId: Long, questionId: Long): Boolean
}
